libcloudforensics/aws.py

```diff
@@ -420,12 +427,19 @@ def CreateVolumeFromSnapshot(self,
 
     client = self.ClientApi(EC2_SERVICE)
     try:
-      volume = client.create_volume(
-          AvailabilityZone=snapshot.availability_zone,
-          SnapshotId=snapshot.snapshot_id,
-          TagSpecifications=[GetTagForResourceType('volume', volume_name)])
+      create_volume_args = {
+          'AvailabilityZone': snapshot.availability_zone,
+          'SnapshotId': snapshot.snapshot_id,
+          'TagSpecifications': [GetTagForResourceType('volume', volume_name)]
+      }
+      if kms_key_id:
+        create_volume_args['Encrypted'] = True
+        create_volume_args['KmsKeyId'] = kms_key_id
+
```

Comment:
You can (probably) move this code out of the try block? Unless `GetTagForResource` type can also raise `ClientError` or `WaiterError`